plenoptic/simulate/canonical_computations/steerable_pyramid_freq.py

```diff
@@ -346,29 +343,18 @@ def forward(self, x, scales=[]):
                     anglemask = self._anglemasks[i][b]
 
                     # bandpass filtering
-                    banddft = lodft * anglemask * himask
-                    banddft = torch.unbind(banddft, -1)
-                    # (x+yi)(u+vi) = (xu-yv) + (xv+yu)i
                     complex_const = np.power(np.complex(0, -1), self.order)
-                    banddft_real = complex_const.real * banddft[0] - complex_const.imag * banddft[1]
-                    banddft_imag = complex_const.real * banddft[1] + complex_const.imag * banddft[0]
-                    # preallocation and then filling in is much more
-                    # efficient than using stack
-                    banddft = torch.empty((*banddft_real.shape, 2), device=banddft_real.device)
-                    banddft[..., 0] = banddft_real
-                    banddft[..., 1] = banddft_imag
-
-                    band = batch_ifftshift(banddft)
-                    band = torch.ifft(band, signal_ndim=2, normalized=self.fft_normalize)
+                    banddft = complex_const * lodft * anglemask * himask
+                    band = fft.ifftshift(banddft)
+                    band = fft.ifft2(band, dim=(-2,-1), norm=self.fft_norm)
+                    
                     if not self.is_complex:
-                        band = torch.unbind(band, -1)[0]
-                        self.pyr_coeffs[(i, b)] = band
-                        self.pyr_size[(i, b)] = tuple(band.shape[-2:])
+                        pyr_coeffs[(i, b)] = band.real
                     else:
                         if self.tight_frame:
```

Comment:
Potentially condition on self.fft_norm instead

Reply:
done in code